From 4db7706e0e904a2b0daaf48ab85c5b156cd45bec Mon Sep 17 00:00:00 2001
From: Richard Kiene <richard.kiene@joyent.com>
Date: Fri, 17 Nov 2017 00:28:43 +0000
Subject: [PATCH] joyent/manta-thoth#156 64bit node binaries are not debuggable
 by mdb when a 32bit node binary is in PATH Reviewed by: Bryan Cantrill
 <bryan@joyent.com> Approved by: Bryan Cantrill <bryan@joyent.com>

---
 bin/thoth    | 8 +++++++-
 package.json | 4 ++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/bin/thoth b/bin/thoth
index 2ac3141..3e3e5cc 100755
--- a/bin/thoth
+++ b/bin/thoth
@@ -1777,7 +1777,13 @@ handlers.debug = function (client, argv)
 		if (analyzer) {
 			cmd += analyzerCmd(args);
 		} else {
-			cmd += 'mdb $MANTA_INPUT_FILE';
+			/*
+			 * LIBPROC_INCORE_ELF=1 prevents us from loading
+			 * whatever node binary happens to be in $PATH. This is
+			 * necessary when debugging cores that do not match the
+			 * bitness of the node binary found in $PATH.
+			 */
+			cmd += 'LIBPROC_INCORE_ELF=1 mdb $MANTA_INPUT_FILE';
 		}
 
 		setAnalyzer(client, cmd, function (asset) {
diff --git a/package.json b/package.json
index 54eb376..2bfff14 100644
--- a/package.json
+++ b/package.json
@@ -1,8 +1,8 @@
 {
 	"name": "manta-thoth",
 	"description": "Manta-based system for core and crash dump analysis",
-	"version": "1.0.0",
-    "license": "MIT",
+	"version": "1.1.0",
+        "license": "MIT",
 	"author": "Joyent (joyent.com)",
 	"dependencies": {
 		"manta": "^4.0.0",
-- 
2.21.0

